src/encryption/update.go

```diff
@@ -0,0 +1,50 @@
+package encryption
+
+import (
+	"fmt"
+
+	"github.com/iancoleman/orderedmap"
+)
+
+func updateContentElement(element interface{}, keyPath path, value interface{}) interface{} {
+
+	switch currentElement := element.(type) {
+	case *orderedmap.OrderedMap:
+		childStep := keyPath[0].String()
+		childElement, ok := currentElement.Get(childStep)
+		if !ok {
+			panic(fmt.Errorf("orderedMap \"%v\" missing key %v", element, childStep))
+		}
+		newChildElement := updateContentElement(childElement, keyPath[1:], value)
+		currentElement.Set(childStep, newChildElement)
+		return currentElement
+	case orderedmap.OrderedMap:
+		childStep := keyPath[0].String()
+		childElement, ok := currentElement.Get(childStep)
+		if !ok {
+			panic(fmt.Errorf("orderedMap \"%v\" missing key %v", element, childStep))
+		}
+		newChildElement := updateContentElement(childElement, keyPath[1:], value)
+		currentElement.Set(childStep, newChildElement)
+		return currentElement
+	case []interface{}:
+		childStep := keyPath[0].(sliceStep)
+		if int(childStep) >= len(currentElement) {
+			panic(fmt.Errorf("slice \"%v\" index %v out of bouds", element, childStep))
+		}
+		childElement := currentElement[childStep]
+		newChildElement := updateContentElement(childElement, keyPath[1:], value)
+		currentElement[childStep] = newChildElement
+		return currentElement
+	// scalar value
+	default:
+		if len(keyPath) > 0 {
+			panic(fmt.Errorf("scalar value update with non-empty path found %v ", keyPath))
```

Comment:
teoreticky to nemusi byt scalar value, skor by si potreboval vediet, ak to nastane (co by teda nemalo) aky je tam typ,
co tak napr.

```
fmt.Errorf("unexpected value "%v" (%T) at path "%s"`, element, element, keyPath)
```

Reply:
ok, dava to lepsi zmysel ked sa upravi ako navrhujes, urobim. btw co by tam este mohlo byt ak nie scalar? Obecne samozrejme akykolvek typ, ale v ramci OrderedMap predpokladam ze by tam mal byt uz len scalar.

Reply:
no:
1. ak to mame spravne, tak to nemoze nastat a nebude tam ani scalar
2. ak to nemame spravne tak do `orderedmap` mozes nastavit lubovolnu hodnot (`interface{}`), takze da sa tam nastavit aj `*FooBar` struct, ... ale v tomto projekte by sa nic take stat nemalo ..... ale ak by bol napr. bug v `blocks` a namiesto vnorenej `orderedmap` by sa tam vlozila struktura `*model.Block`, tak prave tu to spadne, ... a ak nam klient posle log, tak by bolo dobre vediet co za hodnotu sa tam dostala -> skor najdeme pricinu.

Reply:
Konkretne, ak by som na tento kod zabudol (a nebol by ani test), tak sa tam ten `*model.Block` do `content *orderedmap` dostane:
https://github.com/keboola/keboola-as-code/blob/ac16b713bae5f5016f7c3e270b741bbb6e37537d/src/transformation/load.go#L47

Reply:
aha jasne ze tam moze byt bug tak to narazi na non-skalar hodnotu. Inac kedze u toho `default` case sa "zastavi" rekurzia a uz to len prebubla naspat hore a ponastavu sa vratene hodnoty, tak to vlastne umozni nastavit aj takuto non-scalar hodnotu, samozrejme musela by byt v ceste posledna(pradzna keyptah).